tensorboard/util/test_util.py

```diff
@@ -157,12 +159,10 @@ def _impl(reason=None):
 
   return _impl
 
-# TODO(#1996): Better detect TF2.0.
-_is_tf2 =  tf.__version__.startswith('2.')
```

Comment:
I copied this from tensorboard/summary/tf_summary_test.py. May want similar treatment there (tensorboard/compat/__init__.py also has that line btw)

Reply:
It looks to me like `tensorboard/compat/__init__.py` is correct: it’s
checking for the TF2 API, not TF2 behavior. Try running

```
TF2_BEHAVIOR=1 python -c 'import tensorflow as tf; print(tf.constant(1)); tf.summary.create_file_writer'
```

in both V1 and V2, and you’ll see that while both print out an eager
tensor, the V1 environment will fail to deref `create_file_writer`,
while the V2 environment will succeed.

Granted, it’s not exactly crystal-clear, but I’ll err on leaving this
alone to minimally unbreak the build unless we see a reason not to.

Reply:
:+1:  Sounds good

Reply:
As William mentioned, the other places check the API symbols directly because that's what they care about, rather than about whether we're actually under TF 2.0 behavior or not.  In fact those helpers are designed to fall back to `tf.compat.v2` if we aren't actually running TF 2.0 - which is the 2.0 API with 1.x behavior - because they're testing the API surface only.